motor_control/throttle/button_control.ino

```diff
@@ -0,0 +1,36 @@
+#include "button_control.h"
+
+ButtonThrottle::ButtonThrottle(const int &pin, int timeToFull): Throttle(pin, true){
+    this->m_timeToFull = timeToFull;
+    m_isPressed = false;
+}
+
+int ButtonThrottle::mappingFunction(int val, m_map_type mapType){
+  if(mapType = linear){
+    return std::min(val/m_timeToFull, 1) * 1023;
```

Comment:
This is really neat. I think we should maybe have the throttle output range go from [0, 100] though to be more hardware-independent, in case we ever move to another microcontroller that has a different precision for voltage readings. Again, information like this would be really good to include in the `README` so that consumers of this API know exactly what to expect from your class.